check_source.pl

```diff
@@ -213,8 +212,10 @@ BEGIN
                 $diff = diff "$old/$changes", "$dir/$changes";
             }
             for my $line (split(/\n/, $diff)) {
-                next unless $line =~ m/^\+/;
-                $line =~ s/^\+//;
+                # Check also if changes do not revert additional of patch
```

Comment:
typo additional instead of addition ( but feel free to correct my grammar )

Reply:
Apart from the grammar, I miss a comment with some explanation that could prevent this from happening again in the future. If not an explanation, at least a link to https://github.com/openSUSE/openSUSE-release-tools/issues/2640 or https://github.com/openSUSE/openSUSE-release-tools/pull/2586#issuecomment-923755993

Reply:
What about something like this?

```python
# Check whether the line mentions a patch being added (starts with +) or removed (starts with -)
next unless $line =~ m/^[+-]/;
# If any of those cases, remove the patch from the list
$line =~ s/^[+-]//;
```